sanic/app.py

```diff
@@ -206,7 +206,12 @@ def websocket(self, uri, host=None, strict_slashes=False):
         def response(handler):
             async def websocket_handler(request, *args, **kwargs):
                 request.app = self
-                protocol = request.transport.get_protocol()
+                try:
+                    protocol = request.transport.get_protocol()
```

Comment:
I don't think `getattr(request.transport, 'get_protocol()`)` would work, it will raise `AttributeError: 'Transport' object has no attribute 'get_protocol()'`. It has to be:

```python
get_protocol = getattr(request.transport, 'get_protocol', None)
if get_protocol is not None:
    protocol = get_protocol()
else:
    protocol = request.transport._protocol
```

which is not better.